src/abi/mod.rs

```diff
@@ -0,0 +1,70 @@
+use crate::cargo::{manifest::CargoManifestPath, metadata::CrateMetadata};
+use crate::util;
+use near_sdk::__private::{AbiMetadata, AbiRoot};
+use std::collections::HashMap;
+use std::{fs, path::PathBuf};
+
+mod generation;
+
+const ABI_FILE: &str = "abi.json";
+
+/// ABI generation result.
+pub(crate) struct AbiResult {
+    /// Path to the resulting ABI file.
+    pub path: PathBuf,
+}
+
+pub(crate) fn execute(manifest_path: &CargoManifestPath) -> anyhow::Result<AbiResult> {
+    let tmp_dir = tempfile::Builder::new()
+        .prefix("cargo-contract_")
+        .tempdir()?;
```

Comment:
What do you think about persisting the code generated ABI builder source? I, for one, would like to be able to inspect it.

Especially since we're reusing the same target directory of the contract to host the build artifacts of the code generated ABI builder. So we'll have something of the format:

```console
hello-world
├── Cargo.toml
├── src
│   └── lib.rs
└── target
    ├── near
    │   └── hello-world
    │       ├── build.rs
    │       ├── Cargo.toml
    │       ├── src
    │       │   └── lib.rs
    │       └── target
    │           ├── abi.json
    │           └── release
    │               └── hello-world
    └── release
        └── libhello_world.so
```

Reply:
Hmmm, yeah I don't see any immediate downsides to this and this is definitely good for transparency and debugging. I'll try this out.